python/lsst/cp/testing/cp.py

```diff
@@ -0,0 +1,190 @@
+        doc="Dummy exposure.",
+        storageClass='Exposure',
+        dimensions=("instrument", "exposure", "detector"),
+        multiple=True,
+        deferLoad=True,
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        doc="Camera Geometry definition.",
+        storageClass="Camera",
+        dimensions=("instrument", ),
+        isCalibration=True,
+    )
+    inputPtc = cT.Input(
+        name="cptPtc",
+        doc="Input PTC dataset.",
+        storageClass="PhotonTransferCurveDataset",
+        dimensions=("instrument", "detector"),
+        isCalibration=True,
+    )
+
+    outputBFK = cT.Output(
+        name="cptBrighterFatterKernel",
+        doc="Output measured brighter-fatter kernel.",
+        storageClass="BrighterFatterKernel",
+        dimensions=("instrument", "detector"),
+        isCalibration=True,
+    )
+
+
+class CptBrighterFatterKernelSolveConfig(cpPipe.BrighterFatterKernelSolveConfig,
+                                         pipelineConnections=CptBrighterFatterKernelSolveConnections):
+    pass
+
+
+class CptBrighterFatterKernelSolveTask(cpPipe.BrighterFatterKernelSolveTask):
+
+    ConfigClass = CptBrighterFatterKernelSolveConfig
+    _DefaultName = "cptBfkTask"
+
+    pass
+
+
+class CptLinearitySolveConnections(pipeBase.PipelineTaskConnections,
+                                   dimensions=("instrument", "detector")):
+    dummy = cT.Input(
+        name="raw",
+        doc="Dummy exposure.",
+        storageClass='Exposure',
+        dimensions=("instrument", "exposure", "detector"),
+        multiple=True,
+        deferLoad=True,
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        doc="Camera Geometry definition.",
+        storageClass="Camera",
+        dimensions=("instrument", ),
+        isCalibration=True,
+    )
+    inputPtc = cT.Input(
+        name="ptc",
+        doc="Input PTC dataset.",
+        storageClass="PhotonTransferCurveDataset",
+        dimensions=("instrument", "detector"),
+        isCalibration=True,
+    )
+    inputPhotodiodeData = cT.PrerequisiteInput(
+        name="photodiode",
+        doc="Photodiode readings data.",
+        storageClass="IsrCalib",
+        dimensions=("instrument", "exposure"),
+        multiple=True,
+        deferLoad=True,
+        minimum=0,
+    )
+    inputPhotodiodeCorrection = cT.Input(
+        name="pdCorrection",
+        doc="Input photodiode correction.",
+        storageClass="IsrCalib",
+        dimensions=("instrument", ),
+        isCalibration=True,
+    )
+
+    outputLinearizer = cT.Output(
+        name="cptLinearity",
+        doc="Output linearity measurements.",
+        storageClass="Linearizer",
+        dimensions=("instrument", "detector"),
+        isCalibration=True,
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+        if config.applyPhotodiodeCorrection is not True:
+            self.inputs.discard("inputPhotodiodeCorrection")
```

Comment:
Here (and elsewhere) we can do `del inputPhotodiodeCorrection` nowadays.

Also, I prefer `if not config.applyPhotodiodeCorrection:`